frontend/src/metabase/dashboard/actions/tabs.unit.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { TEST_DASHBOARD_STATE } from "../components/DashboardTabs/utils";
+import { moveTab, tabsReducer } from "./tabs";
+
+/**
+ * It's prefererd to write tests in `DashboardTabs.unit.spec.tsx`,
+ * only write tests here for things that are not easily testable at the component level or in Cypress.
+ */
+describe("tabsReducer", () => {
+  it("should reorder the tabs when MOVE_TAB is dispatched", () => {
```

Comment:
Ideally, we would write an e2e test that actually tests the drag and drop behavior. However, [dnd-kit uses pointer events](https://docs.dndkit.com/api-documentation/sensors/pointer) for drag and drop making it very difficult to mock in a test. I think testing the reducer case is a decent compromise that allows us to get some confidence in the correctness of the functionality without requiring a ton of work and possibly introducing flakes.

Reply:
Yeah this is the approach that the library itself uses for its tests https://github.com/clauderic/dnd-kit/blob/5c58f0fe5d19b5aaa5cf93572f3435f4a0a6e54f/cypress/support/commands.ts#L45

I didn't go through the trouble of implementing these long helpers for this PR as I think it's overkill for just this one feature, but if we were to use dnd-kit in other places (and possibly refactor older code to use it) I think it would be worth doing.